src/storage/mvcc/txn.rs

```diff
@@ -533,6 +533,41 @@ impl<S: Snapshot> MvccTxn<S> {
         Ok(is_pessimistic_txn)
     }
 
+    pub fn refresh_lock(&mut self, key: Key, options: &Options) -> Result<u64> {
+        if let Some(lock) = self.reader.load_lock(&key)? {
+            if lock.ts != self.start_ts {
+                // locked by another transaction
+                info!(
+                    "locked by another transaction!";
+                    "previous" => self.start_ts,
+                    "current" => lock.ts,
+                );
+
+                return Err(Error::KeyIsLocked {
+                    key: key.to_raw()?,
+                    primary: lock.primary,
+                    ts: lock.ts,
+                    ttl: lock.ttl,
+                    txn_size: options.txn_size,
+                });
+            }
+            // does not consider pessimistic lock for now.
+            self.lock_key(key, lock.lock_type, lock.primary, lock.short_value, options);
```

Comment:
Refresh lock should only update lock's TTL and other fields should be unchanged. Please set `options.for_update_ts` and `options.txn_size` to lock's.